helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -197,18 +197,17 @@ private void refreshTargetExternalViews(final HelixDataAccessor accessor) {
   public void refreshCustomizedViewMap(final HelixDataAccessor accessor) {
     // As we are not listening on customized view change, customized view will be
     // refreshed once during the cache's first refresh() call, or when full refresh is required
-    List<String> newStateTypes = accessor.getChildNames(accessor.keyBuilder().customizedViews());
     if (_propertyDataChangedMap.get(HelixConstants.ChangeType.CUSTOMIZED_VIEW).getAndSet(false)) {
-      for (String stateType : newStateTypes) {
+      for (String stateType : _aggregationEnabledTypes) {
```

Comment:
This does not look right. If the config types have been changed just before the refresh with fewer items, we won't remove the extra types, right?

Reply:
In refresh logic, the refresh of customized state config (aggregationEnableTypes) happened before customized view. If we read stale aggregationEnabledTypes, it means that the event of customized state config change has not been processed yet. And it should be fine we do not remove stale types, right?

Reply:
I think I misread. This is for the in-memory cache not for the znodes.